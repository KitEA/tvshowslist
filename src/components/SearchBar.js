import React from "react";
import PropTypes from 'prop-types';

const SearchBar = ({ searchBarValue, searchByColumn }) => {
  return (
    <div>
      Search:{" "}
      <input
        type="text"
        placeholder="input value to search"
        value={searchBarValue}
        onChange={searchByColumn}
      />
    </div>
  );
};

SearchBar.propTypes = {
  searchBarValue: PropTypes.string.isRequired,
  searchByColumn: PropTypes.func.isRequired
}

export default SearchBar;