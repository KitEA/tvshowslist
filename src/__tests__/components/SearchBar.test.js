import React from "react";
import renderer from "react-test-renderer";
import SearchBar from "../../components/SearchBar";

describe("SearchBar", () => {
  it("matches the snapshot", () => {
    const tree = renderer
      .create(<SearchBar searchBarValue={""} searchByColumn={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
