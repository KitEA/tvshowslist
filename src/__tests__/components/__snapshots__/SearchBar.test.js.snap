// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar matches the snapshot 1`] = `
<div>
  Search:
   
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="input value to search"
    type="text"
    value=""
  />
</div>
`;
